regional/main.tf

```diff
@@ -401,3 +401,106 @@ resource "kubernetes_manifest" "istio_gateway_mci" {
     }
   }
 }
+
+resource "kubernetes_manifest" "istio_gateway_ca_certificate" {
+  count = var.enable_istio_gateway ? 1 : 0
+
+  manifest = {
+    apiVersion = "cert-manager.io/v1"
+    kind       = "Certificate"
+
+    metadata = {
+      name      = "istio-gateway-ca"
+      namespace = "istio-ingress"
+    }
+
+    spec = {
+      commonName = "istio-gateway-ca"
+      duration   = "2160h"
+      isCA       = true
+
+      issuerRef = {
+        name  = "selfsigned"
+        kind  = "Issuer"
+        group = "cert-manager.io"
+      }
+
+      secretName = "istio-gateway-ca"
+
+      subject = {
+        organizations = ["istio.osinfra.io"]
+      }
+    }
+  }
+}
+
+resource "kubernetes_manifest" "istio_gateway_ca_issuer" {
+  count = var.enable_istio_gateway ? 1 : 0
+  manifest = {
+    apiVersion = "cert-manager.io/v1"
+    kind       = "Issuer"
+
+    metadata = {
+      name      = "istio-gateway-ca"
+      namespace = "istio-ingress"
+    }
+
+    spec = {
+      ca = {
+        secretName = "istio-gateway-ca"
+      }
+    }
+  }
+}
+
+resource "kubernetes_manifest" "istio_gateway_tls" {
+  count = var.enable_istio_gateway ? 1 : 0
+  manifest = {
+    apiVersion = "cert-manager.io/v1"
+    kind       = "Certificate"
+
+    metadata = {
+      name      = "istio-gateway-tls"
+      namespace = "istio-ingress"
+    }
+
+    spec = {
+      commonName = "istio-gateway.osinfra.io"
+      dnsNames   = ["*"]
```

Comment:
I believe I've mentioned this to you before. This is the certificate for TLS termination from the GCLB to the Istio gateway. I'm not sure of another way to do this since we don't know if GCLB is using a hostname or IP. If you have any better suggestions, let me know; otherwise, I'll just leave it.

Reply:
Yes